src/integration/kotlin/SnapshotSenderIntegrationTest.kt

```diff
@@ -0,0 +1,255 @@
+                        linesDone++
+                        logger.info("Checking line $linesDone/$expectedLineCount in $expectedFile")
+                        val jsonLine = parseJson(line)
+                        jsonLine["timestamp"].shouldBe(expectedTimestamp)
+                    } else if (linesDone == expectedLineCount) {
+                        logger.info("Skipping blank line at EOF as should be end of file: have processed $linesDone/$expectedLineCount in $expectedFile")
+                    } else {
+                        fail("Did not expect blank line before EOF: have only processed $linesDone/$expectedLineCount in $expectedFile")
+                    }
+                } while (line != null)
+
+                if (linesDone != expectedLineCount) {
+                    fail("Did get expected line count: have only processed $linesDone/$expectedLineCount in $expectedFile")
+                }
+            }
+        }
+    }
+
+    private fun parseJson(line: String?): JsonObject {
+        try {
+            val stringBuilder = StringBuilder(line)
+            return jsonParser.parse(stringBuilder) as JsonObject
+        }
+        catch (ex: Exception) {
+            fail("Could not parse json line: Got '$ex' from parsing '$line' ")
+        }
+    }
+
+    private fun getFileKeys(fileKeys: NodeList): MutableList<String> {
+        val allKeys = mutableListOf<String>()
+        for (index in 0 until fileKeys.length) {
+            val key = fileKeys.item(index)
+            val keyText = key.textContent
+            logger.info("keyText: $keyText")
+            allKeys.add(keyText)
+        }
+        return allKeys
+    }
+
+    private fun getS3Content(s3FullPath: String): String {
+        val results = Request.Get(s3FullPath)
+            .connectTimeout(1000)
+            .socketTimeout(1000)
+            .execute().returnContent().asString()
+        logger.info("S3 results for '$s3FullPath':\n$results")
+        if (results == null) {
+            throw RuntimeException("No results found for $s3FullPath")
+        }
+        return results
+    }
+
+    private fun getXmlNodesByTagName(keyName: String, sourceXmlString: String): NodeList {
+        val xmlInput = InputSource(StringReader(sourceXmlString))
+        val doc = dBuilder.parse(xmlInput)
+        val keys = doc.getElementsByTagName(keyName)
+            ?: throw RuntimeException("No elements found for '$keyName' in given xml")
+        logger.info("Found ${keys.length} keys with tag name '$keyName'")
+        return keys
+    }
+
+    private fun deriveCollection(fileName: String): String {
+        //s3 in:    test/output/db.core.addressDeclaration-000001.txt.bz2.enc
+        //out       db.core.addressDeclaration
+
+        val lastSlashIndex = fileName.lastIndexOf("/")
+        if (lastSlashIndex < 0) {
+            val errorMessage = "Rejecting: '$fileName' as fileName does not contain '/' to find collection"
+            logger.error(errorMessage)
```

Comment:
this and line 248 are the only places logging at error level, everywhere else seems to simply throw the exception (which I assume gets caught and logged). Is there a reason to deviate and do it differently here?

Reply:
Done